### DISCUSSION TITLE
multiple distributed servers usecase

### DISCUSSION THREAD

**Discussion by aon108** [hibiken/asynq#342] [Q&A] [answered]
thanks for all the great work on this module. very neat and powerful.
I am implementing this as a microservices choreograph/orchestrator, and just wanted to make sure I am using asynq correctly.
each MS will have a server and client running. Server listens on the ms input queue and does work accordingly.
To message other MS, the asynq client posts to the respective task queue.
I cant see any problems with this implementation, but would be great to get your thoughts.
Thank you

**Comment by hibiken** [comment, marked as answer]
@aon108 Thank you for opening this discussion thread!

Running multiple servers to process tasks from the same queue is safe. 
For example, if you have serverA and serverB both configured to consume tasks from Queue1, if you enqueue a task in Queue1, either serverA or serverB will start the work (but not both). One caveat is that you cannot specify which server to process the task when you enqueue it (because both serverA and serverB is waiting for tasks in the same queue, whichever is available first will pick up the task)

**Reply by aon108** [reply]
perfect thank you for getting back to me so quick.

**Comment by chaegumi** [comment]
What should I do in this situation? ServerA and serverB are connected to the same queue at the same time, but I only want serverB to process it, not serverA.
